x-pack/plugins/ml/common/types/annotations.ts

```diff
@@ -60,6 +60,17 @@
 
 import { ANNOTATION_TYPE } from '../constants/annotations';
 
+export type AnnotationFieldName = 'partition_field_name' | 'over_field_name' | 'by_field_name';
+export type AnnotationFieldValue = 'partition_field_value' | 'over_field_value' | 'by_field_value';
+
+export function getAnnotationFieldName(fieldType: string): AnnotationFieldName {
+  return `${fieldType}_name` as AnnotationFieldName;
+}
+
+export function getAnnotationFieldValue(fieldType: string): AnnotationFieldValue {
+  return `${fieldType}_value` as AnnotationFieldValue;
+}
```

Comment:
`fieldType` could not be any string, would be better to add a type. https://github.com/elastic/kibana/blob/35302ed2731404bb2697029314d7bb45b0c49a6b/x-pack/plugins/ml/server/models/results_service/get_partition_fields_values.ts#L12 you can extract this one